arrow-core-data/src/main/kotlin/arrow/core/computations/nullable.kt

```diff
@@ -0,0 +1,27 @@
+package arrow.core.computations
+
+import arrow.continuations.generic.DelimContScope
+import arrow.continuations.generic.DelimitedScope
+
+@Suppress("ClassName")
+object nullable {
+  operator fun <A> invoke(func: suspend BindSyntax.() -> A?): A? =
+    DelimContScope.reset { NullableBindSyntax(this).func() }
+}
+
+
+data class NullableBindSyntax<R>(
+  private val scope: DelimitedScope<R?>,
+) : BindSyntax {
+  override suspend fun <A> A?.bind(): A =
+    scope.shift { cont ->
+      this@bind?.let { cont(it) }
+    }
+}
+
+interface BindSyntax {
+  suspend fun <A> A?.bind(): A
+
+  suspend operator fun <A> A?.not(): A = bind()
+}
```

Comment:
```suggestion
//this will become interface fun when they support suspend in the next release
interface BindSyntax { 
  suspend operator fun <A> A?.invoke(): A 
}
```

Reply:
Extended the comment with a //TODO just to allow discovery on the next release.